tests/testthat/test-translate-mssql.r

```diff
@@ -1,183 +1,95 @@
-  )
-})
-test_that("ceil() translates to CEILING ", {
-  expect_equivalent(
-    translate_sql(ceil(field_name),
-                  con = simulate_mssql()),
-    sql("CEILING(`field_name`)")
-  )
-})
-test_that("trimws() translates to LTRIM-RTRIM ", {
-  expect_equivalent(
-    translate_sql(trimws(field_name),
-                  con = simulate_mssql()),
-    sql("LTRIM(RTRIM(`field_name`))")
-  )
+  expect_equal(trans(sd(x, na.rm = TRUE)),  sql("STDEV(`x`)"))
+  expect_equal(trans(var(x, na.rm = TRUE)), sql("VAR(`x`)"))
+
+  expect_error(trans(cor(x)), "not available")
+  expect_error(trans(cov(x)), "not available")
 })
 
-# stringr -------------------------------------------
+# MSSQL window conversions -----------------------------------------
+test_that("custom aggregators translated correctly", {
 
-test_that("str_length() translates correctly ", {
-  expect_equivalent(
-    translate_sql(
-      str_length(field_name),
-      con = simulate_mssql()),
-    sql("LEN(`field_name`)"))
-})
+  trans <- function(x) {
+    translate_sql(!!enquo(x), window = TRUE, con = simulate_mssql())
+  }
 
-test_that("str_locate() translates correctly ", {
-  expect_equivalent(
-    translate_sql(
-      str_locate(field_name, "find"),
-      con = simulate_mssql()),
-    sql("CHARINDEX('find', `field_name`)"))
-})
+  expect_equal(trans(sd(x, na.rm = TRUE)),  sql("STDEV(`x`) OVER ()"))
+  expect_equal(trans(var(x, na.rm = TRUE)), sql("VAR(`x`) OVER ()"))
 
-test_that("str_detect() translates correctly ", {
-  expect_equivalent(
-    translate_sql(
-      str_detect(field_name, "find"),
-      con = simulate_mssql()),
-    sql("CHARINDEX('find', `field_name`) > 0"))
+  expect_error(trans(cor(x)), "not supported")
+  expect_error(trans(cov(x)), "not supported")
 })
 
-
 # MSSQL query tests  ------------------------------------------------
+test_that("filter and mutate translate is.na correctly", {
+  mf <- lazy_frame(x = 1, src = simulate_mssql())
 
-df <- data.frame(x = 1, y = 2)
-df_mssql <- tbl_lazy(df, src = simulate_mssql())
+  expect_equal(
+    mf %>% head() %>% show_query(),
+    sql("SELECT  TOP 6 *\nFROM `df`")
+  )
 
-test_that("query uses TOP instead of LIMIT ", {
-  expect_equivalent(
-    show_query(head(df_mssql)),
-    sql("SELECT  TOP 6 *\nFROM `df`"))
-})
+  expect_equal(
+    mf %>% mutate(z = is.na(x)) %>% show_query(),
+    sql("SELECT `x`, CONVERT(BIT, IIF(`x` IS NULL, 1, 0)) AS `z`\nFROM `df`")
+  )
 
-test_that("Mutate uses CONVERT-BIT for is.na ", {
-  expect_equivalent(
-    show_query(mutate(df_mssql, z = is.na(x))),
-    sql("SELECT `x`, `y`, CONVERT(BIT, IIF(`x` IS NULL, 1, 0)) AS `z`\nFROM `df`"))
-})
+  expect_equal(
+    mf %>% filter(is.na(x)) %>% show_query(),
+    sql("SELECT *\nFROM `df`\nWHERE (((`x`) IS NULL))")
+  )
 
-test_that("Filter uses IS NULL for is.na ", {
-  expect_equivalent(
-    show_query(filter(df_mssql, is.na(x))),
-    sql("SELECT *\nFROM `df`\nWHERE (((`x`) IS NULL))"))
-})
+  expect_equal(
```

Comment:
I don't think you need the tests of `!is.na()` here - it's just testing the composition of `!` and `is.na()`

Reply:
I think you missed this one?

Reply:
I just tested that query against an open MS SQL connection and it failed.  I need to run more "live" tests with a negated `is.na()`.  Is it ok for me to include the needed translations changes in this PR or should I do a separate one?